libraries/node-core-library/src/JsonFile.ts

```diff
@@ -146,17 +146,35 @@ export class JsonFile {
    */
   // tslint:disable-next-line:no-any
   public static validateNoUndefinedMembers(jsonObject: Object): void {
+    return JsonFile._validateNoUndefinedMembers(jsonObject, '');
+  }
+
+  // Private implementation of validateNoUndefinedMembers()
+  private static _validateNoUndefinedMembers(jsonObject: Object, path: string): void {
     if (!jsonObject) {
       return;
     }
     if (typeof jsonObject === 'object') {
       for (const key of Object.keys(jsonObject)) {
+        let fullPath: string = path;
+        if (Array.isArray(jsonObject)) {
+          fullPath += `[${key}]`;
+        } else if (/^[a-z_0-9]+$/i.test(key)) {
+          if (fullPath) {
+            fullPath += '.';
+          }
+          fullPath += `${key}`;
+        } else {
+          fullPath += `["${key.replace(/["]/g, '\\"')}"]`;
+        }
+
```

Comment:
can you add a comment? there are a lot of symbols here, and not everyone wants to read the regex

Reply:
I separated it into another function add added documentation.  I also improved perf by only performing the formatting is if an error is detected

---
In reply to: [137663256](https://github.com/Microsoft/web-build-tools/pull/363#discussion_r137663256) [](ancestors = 137663256)